lib/netsuite/support/records.rb

```diff
@@ -49,7 +49,15 @@ def to_attributes!(hash, kname, v)
       end
 
       def record_type
-        "#{record_namespace}:#{self.class.to_s.split('::').last}"
+        "#{record_namespace}:#{record_type_without_namespace}"
+      end
+
+      def type
+        record_type_without_namespace.downcase
+      end
```

Comment:
@jdedels I'm not too familiar with the `attach_file` call, but I believe types should be lower camel case, i.e. `customer`, `salesOrder`, `transferOrder`, etc. Do you know if `attach_file` uses a different type case or did you just test this with single-word objects?

Thanks again for your help here! Sorry about the delay in taking a look at this